target/linux/ath79/image/generic.mk

```diff
@@ -1813,6 +1813,21 @@ define Device/hiwifi_hc6361
 endef
 TARGET_DEVICES += hiwifi_hc6361
 
+define Device/huawei_ap5030dn
+  SOC := qca9550
+  DEVICE_VENDOR := Huawei
+  DEVICE_MODEL := AP5030DN
+  DEVICE_PACKAGES := ath10k-firmware-qca988x-ct kmod-ath10k-ct
+  LOADER_TYPE := bin
+  LOADER_FLASH_OFFS := 0x111E00
```

Comment:
Add a `KERNEL_SIZE` parameter the size of the first partition so that the kernel does not get split between both partitions.

Reply:
done